pkg/kubectl/cmd/util/config_flags.go

```diff
@@ -0,0 +1,215 @@
+package util
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	"k8s.io/kubernetes/pkg/kubectl/util/transport"
+)
+
+const (
+	FlagClusterName      = "cluster"
+	FlagAuthInfoName     = "user"
+	FlagContext          = "context"
+	FlagNamespace        = "namespace"
+	FlagAPIServer        = "server"
+	FlagInsecure         = "insecure-skip-tls-verify"
+	FlagCertFile         = "client-certificate"
+	FlagKeyFile          = "client-key"
+	FlagCAFile           = "certificate-authority"
+	FlagEmbedCerts       = "embed-certs"
+	FlagBearerToken      = "token"
+	FlagImpersonate      = "as"
+	FlagImpersonateGroup = "as-group"
+	FlagUsername         = "username"
+	FlagPassword         = "password"
+	FlagTimeout          = "request-timeout"
+)
+
+// ConfigFlags composes the set of values necessary
+// for obtaining a REST client config
+type ConfigFlags struct {
+	CacheDir   *string
+	KubeConfig *string
+
+	// config flags
+	ClusterName      *string
+	AuthInfoName     *string
+	Context          *string
+	Namespace        *string
+	APIServer        *string
+	Insecure         *string
+	CertFile         *string
+	KeyFile          *string
+	CAFile           *string
+	EmbedCerts       *string
+	BearerToken      *string
+	Impersonate      *string
+	ImpersonateGroup *string
+	Username         *string
+	Password         *string
+	Timeout          *string
+
+	// DisableUsernamePassword allows for the username and password config
+	// options to be disabled, in turn using a non-interactive client config loader.
+	DisableUsernamePassword bool
```

Comment:
I'd turn this into a struct's private member you can only set once when initiating the struct with `NewConfigFlags`

Reply:
> I'd turn this into a struct's private member you can only set once when initiating the struct with NewConfigFlags

Doing that means you can't construct a custom version of this.  YOu'd have to call the constructor and get a bunch of values already set.